File: src/scala/Palindrome.scala
object Palindrome {
  def isPalindrome(x: Int): Boolean = {
    val isPalindrome = {
      if (x == 0) {
        true
      }else if (x < 0) {
        false
      }else {
        val sb = new StringBuilder(x + "")
        val numberStr = sb.toString
        val reverseNumStr = sb.reverse.toString
        numberStr == reverseNumStr
      }
    }
    isPalindrome
  }
}

File: src/java/org/jy/Palindrome.java
package org.jy;

/**
 * 9、回文数
 * @author LJY
 */
public class Palindrome {

    public boolean isPalindrome(int x) {
        if(x < 0){
            return false;
        } else if(x == 0){
            return true;
        } else {
            StringBuilder sb = new StringBuilder(x + "");
            String numberStr = sb.toString();
            String reverseNumStr = sb.reverse().toString();
            return numberStr.equals(reverseNumStr);
        }

    }

}
